src/main/java/com/codeborne/selenide/appium/DeepLinkLauncher.java

```diff
@@ -0,0 +1,88 @@
+import static io.appium.java_client.ios.options.wda.SupportsAutoAcceptAlertsOption.AUTO_ACCEPT_ALERTS_OPTION;
+
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.logevents.SelenideLogger;
+import com.google.common.collect.ImmutableMap;
+import io.appium.java_client.AppiumDriver;
+import io.appium.java_client.HasOnScreenKeyboard;
+import java.time.Duration;
+import java.util.Map;
+import javax.annotation.ParametersAreNonnullByDefault;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.HasCapabilities;
+import org.slf4j.helpers.CheckReturnValue;
+
+@ParametersAreNonnullByDefault
+public class DeepLinkLauncher {
+  public static Duration SAFARI_ELEMENTS_TIMEOUT = Duration.ofSeconds(30);
+  private static final String SAFARI_BUNDLE_ID = "com.apple.mobilesafari";
+
+  // adopted from https://bit.ly/3OKVsvq
+  // see also https://appiumpro.com/editions/84-reliably-opening-deep-links-across-platforms-and-devices
+  public void openDeepLinkOnIos(AppiumDriver driver, String deepLinkUrl) {
+    SelenideLogger.run("open ios deeplink", deepLinkUrl, () -> {
+      HasOnScreenKeyboard keyboard = cast(driver, HasOnScreenKeyboard.class)
+        .orElseThrow(() -> new IllegalArgumentException("Driver doesn't support HasOnScreenKeyboard"));
+      openSafari(driver);
+      // Add the deep link url in Safari in the `URL`-field
+      // This can be 2 different elements, or the button, or the text field
+      // Use the predicate string because the accessibility label will return 2 different types
+      // of elements making it flaky to use. With predicate string we can be more precise
+      SelenideElement addressBar = $(iOSNsPredicateString("label == 'Address' OR name == 'URL'"));
+      SelenideElement urlField = $(iOSNsPredicateString("type == 'XCUIElementTypeTextField' && name CONTAINS 'URL'"));
+      // Wait for the url button to appear and click on it so the text field will appear
+      // iOS 13 now has the keyboard open by default because the URL field has focus when opening the Safari browser
+      if (!keyboard.isKeyboardShown()) {
+        addressBar.click();
+      }
+      // Submit the url and add a break
+      urlField.shouldBe(visible, SAFARI_ELEMENTS_TIMEOUT).setValue(deepLinkUrl + "\uE007");
+      // if you started the iOS device with `autoAcceptAlerts:true` in the capabilities
+      // then Appium will auto accept the alert that should be shown now.
+      if (!canAutoAcceptAlerts(driver)) {
+        // Wait for the notification and accept it
+        // When using an iOS simulator you will only get the pop-up once, all the other times it won't be shown
+        try {
+          SelenideElement openButton =
+            $(iOSNsPredicateString("type == 'XCUIElementTypeButton' && name CONTAINS 'Open'"));
+          openButton.shouldBe(enabled, SAFARI_ELEMENTS_TIMEOUT).click();
+        } catch (AssertionError e) {
+          // ignore
```

Comment:
1. Maybe we should log this `e`? At lest with `debug` level?
2. Don't we spend 4 seconds and take an unneeded screenshot in this case?
@BorisOsipov

Reply:
1. agree. will fix.
2. we spend 30 secs. That's why I open backdoor [via public timeout](https://github.com/selenide/selenide-appium/blob/7c8ab08ce2820d169978d782fe23826700f22876/src/main/java/com/codeborne/selenide/appium/DeepLinkLauncher.java#L24) to allow user fix it temporary. I have no idea how to disable unneeded screenshot here :(